utils/firestore.js

```diff
@@ -433,6 +433,66 @@ const retrieveParticipantsEligibleForIncentives = async (siteCode, roundType, is
     }
 }
 
+const removeParticipantsDataDestruction = async (totalDocNumber = 10) => {
+    try {
+        const dataDestructionFieldList = ['104278817','119449326','153713899','173836415','231676651','262613359','268665918','269050420','304438543','359404406','399159511','407743866','412000022','471168198','479278368','524352591','526455436','544150384','558435199','577794331','592227431','613641698', '664453818','744604255', '747006172','765336427','773707518','826240317','831041022','883668444','996038075','token','Connect_ID','query','pin'];
+        let pageLimit = 500;
+        let pageCount = 0;
+        let toContinue = true;
+
+        pageLimit = Math.min(pageLimit, totalDocNumber);
+        const { isIsoDate } = require('./validation');
+        while (toContinue) {
+            const batch = db.batch();
+            const checkedDocCount = pageCount * pageLimit;
+            const remainingDocCount = totalDocNumber - checkedDocCount;
```

Comment:
would anything blow up if `remainingDocCount` was a negative number? looking at the variables above, it seems like that could easily happen

Reply:
Solved